<?xml version="1.0"?>
<launch>
  <!--
    (remove backslashes https://stackoverflow.com/questions/10842131/xml-comments-and)
    v4l2-ctl \-\-list-formats
    v4l2-ctl -d /dev/video0 \-\-list-framesizes=YUYV
    -->
  <arg name="use_image_publisher" default="true"/>

  <arg name="use_usb_cam" default="false"/>
  <arg name="device" default="0"/>
  <arg name="cam_width" default="640"/>
  <arg name="cam_height" default="480"/>
  <arg name="cam_frame_rate" default="10"/>

  <!-- TODO(lwalter) the latest usb_cam fails when given an invalid
    width/height/format, the old bosch one made it work
    Maybe it could be fixed to fail through to closest match
    and then change the format on output -->
  <group if="$(arg use_usb_cam)">
  <node name="image_source" pkg="usb_cam" type="usb_cam_node"
      output="screen"
      if="$(arg use_usb_cam_node)">
    <param name="image_width" value="$(arg cam_width)"/>
    <param name="image_height" value="$(arg cam_height)"/>
    <param name="framerate" value="$(arg cam_frame_rate)"/>
    <param name="video_device" value="/dev/video$(arg device)" />
    <!--param name="pixel_format" value="mjpeg" /-->
    <param name="pixel_format" value="yuyv"/>
  </node>
  <node name="v4l2ucp" pkg="v4l2ucp" type="v4l2ucp" >
    <param name="device" value="/dev/video$(arg device)" />
  </node>
  <!-- TODO(lucasw) Need to delay launch of this until the v4l2ucp node above
    has complete creating all the parameters- maybe it should publish an 'update'
    signal? -->
  <node name="dr_topics" pkg="dynamic_reconfigure_tools" type="dr_topics.py" >
  </node>
  </group>  <!-- use_usb_cam -->

  <group ns="test_image_pub/image_raw">
    <rosparam param="disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
  </group>

  <node pkg="image_publisher" type="image_publisher"
    name="test_image_pub"
    args="$(find vimjay)/data/clouds.png"
    if="$(arg use_image_publisher)">
    <param name="publish_rate" value="0.5"/>
  </node>

<!--
  <node pkg="imgui_ros" type=""
    name="load0" >
  </node>
-->

</launch>
